db-engine/src/main/java/com/splicemachine/db/impl/sql/GenericStatement.java

```diff
@@ -619,6 +620,22 @@ else if (newMergeJoinString.equals("forced"))
                 }
                 cc.setCurrentTimestampPrecision(currentTimestampPrecision);
 
+                String timestampPrecisionString =
+                        PropertyUtil.getCachedDatabaseProperty(lcc, Property.SPLICE_TIMESTAMP_PRECISION);
+                int timestampPrecision = CompilerContext.DEFAULT_TIMESTAMP_PRECISION;
+                try {
+                    if (timestampPrecisionString != null)
+                        timestampPrecision = Integer.parseInt(timestampPrecisionString);
+                    if (timestampPrecision < Limits.MIN_TIMESTAMP_PRECISION)
+                        timestampPrecision = Limits.MIN_TIMESTAMP_PRECISION;
+                    if (timestampPrecision > Limits.MAX_TIMESTAMP_PRECISION)
+                        timestampPrecision = Limits.MAX_TIMESTAMP_PRECISION;
+                } catch (Exception e) {
+                    // If the property value failed to convert to a boolean, don't throw an error,
```

Comment:
Better to throw the error so that the user can correct the property values

Reply:
@jyuanca I believe this is not the code to handle the statement itself, but to parse the value once it's already been set.  So, if we throw the Exception here, that would cause all SQL requests to fail after a user has set an out-of-range value for this property.  So, the current code may be OK.

Reply:
@jyuanca does @msirek's address your comment?